app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -0,0 +1,137 @@
+(function ($) {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function OptionSelect(options){
+    /* This JavaScript provides two fucntional enhancements to option-select components:
+      1) A link that clears all of the checkboxes (referred to here as the "clearing link")
+      2) Open/closing of the list of checkboxes - this is not provided for ie6 and 7 as the performance is too janky.
+    */
+
+    this.$optionSelect = options.el;
+    this.$options = this.$optionSelect.find("input[type='checkbox']");
+
+    // Build clearing link
+    this.$clearingLink = this.attachClearingLink();
+    this.updateClearingLink();
+
+    // Attach event listeners for clearing
+    this.$clearingLink.on('click', this.resetOptions.bind(this));
+    this.$options.on('click', this.updateClearingLink.bind(this));
+
+    // Performance in ie 6/7 is not good enough to support animating the opening/closing
+    // so do not allow option-selects to be collapsible in this case
+    var allowCollapsible = (typeof ieVersion == "undefined" || ieVersion > 7) ? true : false;
```

Comment:
We haven't defined `ieVersion` globally only on an app by app basis. If you want to use it in this I would suggest adding them to [the template slimmer uses for all pages](https://github.com/alphagov/static/blob/master/app/views/root/_stylesheet.html.erb) in the [same way they have been added to finders](https://github.com/alphagov/finder-frontend/blob/master/app/views/layouts/application.html.erb#L6-L8)